### PR TITLE
Fix for NodeLabel plugin

### PR DESCRIPTION
**Pull request by rsandell**
Sent to me via mail

> I'm not familiar with GitHub, but when using Rebuild plugin with NodeLabel plugin, I have been facing some issues.
> 
> I have succeeded to fix it for my own jobs, and I give you my code if you want to deliver it on GitHub, you're welcome.

### CODE REVIEW

File: src/main/java/com/sonyericsson/rebuild/RebuildAction.java [hunk 1]
```diff
@@ -430,9 +435,19 @@ private ParameterValue cloneParameter(ParameterValue oldValue, String newValue)
...
             return new StringParameterValue(oldValue.getName(), newValue, oldValue.getDescription());
+        } else if (oldValue.getClass().getName().equals(NODE_LABEL_PARAM_CLASS)) {
+            /**
+             * getClass().getName() to avoid dependency on nodelabel plugin.
+             */
+            try {
+                Constructor ctor = oldValue.getClass().getConstructor(String.class, String.class);
+                return (ParameterValue)ctor.newInstance(new Object[]{oldValue.getName(), newValue});
+            } catch (Exception e) {
+            }
```
Comment (oleg-nenashev) [inline src/main/java/com/sonyericsson/rebuild/RebuildAction.java]: Maybe needs logging

**Comment by orgads** [comment]
This should solve JENKINS-38635. Please merge it.

File: src/main/java/com/sonyericsson/rebuild/RebuildAction.java [hunk 2]
```diff
@@ -67,6 +68,7 @@
 public class RebuildAction implements Action {
 
     private static final String SVN_TAG_PARAM_CLASS = "hudson.scm.listtagsparameter.ListSubversionTagsParameterValue";
+    private static final String NODE_LABEL_PARAM_CLASS = "org.jvnet.jenkins.plugins.nodelabelparameter.LabelParameterValue";
```
Comment (imod) [inline src/main/java/com/sonyericsson/rebuild/RebuildAction.java]: there is also `org.jvnet.jenkins.plugins.nodelabelparameter.NodeParameterValue`, which extends `org.jvnet.jenkins.plugins.nodelabelparameter.LabelParameterValue`

**Comment by jdsn** [comment]
👍 please merge it, it would make rebuild plugin useful for us again

**Comment by orgads** [comment]
Actually I tried it and it didn't work. Did anyone else verify it?

**Comment by jdsn** [comment]
Oh, too bad.

@rsandell can you please have a look at this PR and change it so it can be merged finally?

**Comment by ejpenney** [comment]
Not sure if this helps, but I noticed that changing this line in NodeParameterValue.jelly:

`            <f:textbox name="labels" value="${it.label}" />
`
and removing the "s" in "labels":
`            <f:textbox name="label" value="${it.label}" />
`
Fixed the issue for me.  Rebuilds of jobs with NodeParameterValues seems to work now.

**Comment by rsandell** [comment]
Well this actually not my change as it says in the description. I only uploaded the patch for the one who sent it to me and gave him the link hoping it wasn't a drive by contribution :'(

**Comment by GLundh** [comment]
I have merged https://github.com/jenkinsci/rebuild-plugin/pull/45 which seems to work fine for me. Any reason to keep this open? @jdsn : Do you want to verify that the issue is fixed on the master branch?

**Comment by GLundh** [comment]
No response for 8 days. Closing this one.

**Pull request closed without merging**